tools/expotools/src/versioning/ios/index.ts

```diff
@@ -782,42 +782,56 @@ exports.addVersionAsync = async function addVersionAsync(
 
   // some files in the Optional spec should be omitted from versioned code
   const excludedOptionalDirectories = getExcludedOptionalDirectories();
-  excludedOptionalDirectories.forEach(dir => {
+
+  for (const dir of excludedOptionalDirectories) {
     const optionalPath = path.join(newVersionPath, 'Expo', 'Optional', dir);
+
     try {
-      fs.accessSync(optionalPath, fs.F_OK);
-      shell.exec(`rm -rf ${optionalPath}`);
+      await fs.access(optionalPath, fs.F_OK);
+      await fs.remove(optionalPath);
     } catch (e) {
       console.warn(`Expected ${optionalPath} to be accessible for removal, but it wasn't`);
     }
-  });
+  }
 
   // Copy universal modules into the clone
   console.log(`Copying universal modules into the new Pods...`);
-  shell.exec(
-    Modules
-      .getVersionableModulesForPlatform('ios', versionNumber)
-      .map(({ libName, podName }) =>
-        [
-          `cp -R ${rootPath}/${RELATIVE_UNIVERSAL_MODULES_PATH}/${libName}/ios/ ${newVersionPath}/${podName}`,
-          `mv ${newVersionPath}/${podName}/${podName} ${newVersionPath}/${podName}/${versionedPodNames[podName]}`,
-          `cp -R ${rootPath}/${RELATIVE_UNIVERSAL_MODULES_PATH}/${libName}/package.json ${newVersionPath}/${podName}`,
-        ].join(' && ')
-      )
-      .join(' && ')
-  );
+  const modules = Modules.getVersionableModulesForPlatform('ios', versionNumber);
```

Comment:
Do we still want to use `xdl/Modules` or would we want to move to `packages` search?

Reply:
I didn't plan to refactor all these stuff right now, it would probably take too much time and I guess it would be better to do after SDK34 is released. Anyway, so far we don't have any replacement for `versionable` property in xdl config - so we wouldn't know which modules shouldn't be versioned. I also don't want to have them hardcoded, so I'll look for another way later.